xdsl/dialects/arith.py

```diff
@@ -589,6 +589,22 @@ def get(operand1: Union[Operation, SSAValue],
         return Minf.build(operands=[operand1, operand2],
                           result_types=[operand1.typ])
 
+@irdl_op_definition
+class IndexCastOp(Operation):
+    name = "arith.index_cast"
+
+    input: Operand
```

Comment:
```suggestion
    input: Annotated[Operand, AnyAttr()]
```

Reply:
I think the recommended way to spell this is `Annotated[Operand, AnyAttr]` without the `()`

Reply:
Not sure whether we changed it already. I am sure we had the discussion lately. I am quite sure though that `()` will work...

Reply:
I don't understand why is this the recommended way. Does it provide us with anything? I thought it's cleaner to remove annotations where they don't add anything...

Reply:
I agree with Anton, we shouldn't add annotations here, because none are needed.
`AnyAttr()` would just add a constraint that's always `True`, so it wouldn't add anything.
I think it's cleaner to have just `Operand`.

Reply:
I agree, but as I argue in the other thread, we should do a conscious decision here, because this is what people will inspect to understand our API and I think we used the Annotated approach in other parts of the repo.